homeassistant/components/diagnostics/__init__.py

```diff
@@ -51,12 +51,12 @@ class DiagnosticsProtocol(Protocol):
 
     async def async_get_config_entry_diagnostics(
         self, hass: HomeAssistant, config_entry: ConfigEntry
-    ) -> dict:
+    ) -> Any:
```

Comment:
I thought of using `dict | list`, I thought of maybe adding `None` but then it could also be anything that is json serializable...

Reply:
Yes, it needs to be json serializable, but otherwise no restrictions.